### PR TITLE
Reconstructing pruning API derived from Component.py.

### PR DESCRIPTION
**Pull request by YIYANGCAI**
Signed-off-by: YIYANGCAI <yiyang.cai@intel.com>

## Type of Change

API changed or not

## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/conf/pythonic_config.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import logging
 from .dotdict import DotDict
 from ..config import _BaseQuantizationConfig, accuracy_criterion, BenchmarkConfig, \
-                     check_value, DistillationConfig, options, PruningConfig
+                     check_value, DistillationConfig, options, WeightPruningConfig
```
Comment (ftian1) [inline neural_compressor/conf/pythonic_config.py:21]: here should we introduce new WeightPruningConfig and keep old PruningConfig as the latter is used by 1.x pruning implementation?

Comment (ftian1) [reply neural_compressor/conf/pythonic_config.py:21]: so pls check if INC 1.14.2 includes PruningConfig or not. if yes, it means 1.x pruning code used this definition before and should keep and add new WeightPruningConfig but not replace?

Comment (wenhuach21) [reply neural_compressor/conf/pythonic_config.py:21]: As you mentioned before, PruningConfig should be only used in 2.0a. I have checked  neural_compressor/config.py and nerual_compressor/conf/config in v1.4.2, this class has not been defined

Comment (wenhuach21) [reply neural_compressor/conf/pythonic_config.py:21]: > here should we introduce new WeightPruningConfig and keep old PruningConfig as the latter is used by 1.x pruning implementation?
As you mentioned before, PruningConfig should be only used in 2.0a. I have checked neural_compressor/config.py and nerual_compressor/conf/config in v1.4.2, this class has not been defined

File: examples/pytorch/nlp/huggingface_models/question-answering/pruning/pytorch_pruner/eager/run_qa_no_trainer.py [hunk 2]
```diff
@@ -57,8 +56,8 @@
 from transformers.utils import check_min_version
 from transformers.utils.versions import require_version
 from utils_qa import postprocess_qa_predictions
-from neural_compressor.pruning import Pruning
-from neural_compressor.pruner.utils import WeightPruningConfig
+from neural_compressor.pruner.pruning import Pruning
```
Comment (ftian1) [inline examples/pytorch/nlp/huggingface_models/question-answering/pruning/pytorch_pruner/eager/run_qa_no_trainer.py:59]: Pruning under pruner is not correct here. let's have quick sync

**Review by ftian1 — APPROVED** [review]

**Pull request merged**